packages/Angular/ask-skip/src/lib/misc/user-view-grid-with-analysis.component.ts

```diff
@@ -38,17 +38,25 @@ import { MJTabStripComponent } from "@memberjunction/ng-tabstrip";
 
     @ViewChild(UserViewGridComponent, {static: false}) viewGrid!: UserViewGridComponent;
     @ViewChild(MJTabStripComponent, {static: false}) tabStrip!: MJTabStripComponent;
+
+    private _compositeKey: CompositeKey = new CompositeKey();
+
     public get ViewID(): number {
         if (this.Params && this.Params.ViewID)
           return this.Params.ViewID;
         else
             return 0;
     }    
     /**
-     * Returns the ViewID as an array of KeyValuePair
+     * Returns the ViewID as a composite key
      */
-    public get ViewIDAsPrimaryKeyArray(): KeyValuePair[] {
-        return [{FieldName: "ID", Value: this.ViewID}];
+    public get ViewIDAsCompositeKey(): CompositeKey {
+        if(this._compositeKey.KeyValuePairs){
+            return this._compositeKey;
+        }
+
+        this._compositeKey.KeyValuePairs = [{FieldName: "ID", Value: this.ViewID}];
```

Comment:
@JS-BC  might be helpful to have a constructor for CompositeKey with optional params taking in an array of PKVals and if provided, it does this so that your code can be to just return `new CompositeKey(pkarray)`

Reply:
Good idea